src/controllers/hiring-process/index.ts

```diff
@@ -1,37 +1,16 @@
 import { getRepository } from 'typeorm'
 import { validate } from 'class-validator'
 import { HiringProcess } from '@models/entity/HiringProcess'
-import { message } from '../../messages/languages/pt-br'
-import { HiringProcessRequest } from '../../service/hiring-process/HiringProcessRequest'
-import { HiringProcessService } from '../../service/hiring-process/HiringProcessService'
-import { HttpResponseHandler } from '@controllers/HttpResponseHandler'
-
-const hiringService = new HiringProcessService()
-const httpResponseHandler = new HttpResponseHandler()
-
-const getStatus = (startDate, endDate) => {
-  const currentDate = Date.now()
-
-  const statusPreparation = currentDate < startDate
-  const statusOpened = currentDate >= startDate && currentDate <= endDate
-
-  if (statusPreparation) {
-    return 'status-preparing'
-  }
-  if (statusOpened) {
-    return 'status-opened'
-  }
-
-  return 'status-closed'
-}
+import { message } from '@messages/languages/pt-br'
+import { createHiringProcessService } from '@service/hiring-process'
+import { createErrorResponse, createSuccessResponse } from '@service/HttpHandler'
 
 export const createHiringProcessEndpoint = async (request, response) => {
   try {
-    const hiringProcessRequest = HiringProcessRequest.convertFromHttpBody(request.body)
-    const result = await hiringService.createHiringProcessService(hiringProcessRequest)
-    return httpResponseHandler.createSuccessResponse(message.SUCCESS, result, response)
+    const result = await createHiringProcessService(request)
```

Comment:
Fiquei na dúvida, mandando a request assim não estaria criando um acoplamento do protocolo com o serviço? penso que seria interessante separar uma request http dos serviços presando pelo baixo acoplamento, assim facilita testes e reutilização por outros protocolos.

Reply:
> Acho que ainda seria interessante passar um pouco do conceito de classes/OO para as alunas, por ser um paradigma amplamente utilizado no mercado. Mas estou percebendo que sem passar uma base do conceito antes, estar um pouco mais complicado agora para ensinar isso na pratica. Dito isso, acho que faz sentido migrar para um paradigma único. Mas minha preocupação ainda fica: Como passar esse conhecimento prático para as alunas? para que em processos seletivos no futuro não seja um ponto que prejudique elas.

Motivação:
Essa é a primeira turma que decidimos não ensinar OO ou Java, acredito que estamos indo bem. Conseguimos manter mais de 50% delas niveladas, e principalmente conseguindo entregas significativas pro projeto e mais adesão. Não foi 1 ou 2 turmas que trouxeram feedback sobre "pouco tempo" ou coisas sobre "java" ser complexo. E quando vc olha pro Java a única coisa que o torna complexo é a necessidade de aprender OO, já que a linguagem é fortemente orientada a objeto. A partir disso resolvemos não ensinar OO ou Java depois desses avanços e principalmente dos muitos feedbacks das turmas anteriores. 

Agora uma opinião minha:
Acho que a gente não precisamos por em prática na aceleradora exatamente tudo que o mercado de trabalho usa, a jornada delas não encerra somente na Aceleradora e temos pouquíssimo tempo para conseguir que todas caminhem na mesma página, adicionar mais complexidade tornaria isso mais difícil. Temos alunas que nunca programaram na vida em menos de 5 meses elas não vão dominar um padrão de projeto e nem muito menos testes unitários (tenho colegas de trabalho que não dominam com mais de 3 anos de experiencia ou nunca trabalharam com testes por exemplo). 

Acho importante elas terem sessões para conhecerem os assuntos e verem as possibilidades, e tentarem por elas implementar o que foi aprendido

Reply:
> Fiquei na dúvida, mandando a request assim não estaria criando um acoplamento do protocolo com o serviço? penso que seria interessante separar uma request http dos serviços presando pelo baixo acoplamento, assim facilita testes e reutilização por outros protocolos.

agora sobre isso, eu tentei refatorar para funções pq é o que elas se sentiam a vontade, mas eu acredito que ainda assim ela sentiriam dificuldade em manipular esse codigo. Não acho que esse PR vai pra frente, na real to com vontade de apresentar pra elas so pq eu prometi, mas acho que o sentimento tanto pra abordagem com OO quando com essa dai vai dar um nó na cabeça delas

Reply:
Acho que faz sentido seus pontos,  a ideia era mais usar pouco de OO com classes no Java Script já que me parece ser bastante utilizado em backend para JavaScript e não ensinar toda a OO e Java como mencionaste tentando  introduzir com um pouco do que existe. Concordo que se as alunas estão com dificuldade precisamos dar um passo atrás mesmo e rever se faz sentido. 

Sinto que causei um pouco de ruído, acho que no fundo todo o processo de criação precisa de um pouco de ruído para uma discussão saudável mas se foi demais sinto muito hehehe